internal/rest/avi_obj_pool.go

```diff
@@ -122,13 +121,18 @@ func (rest *RestOperations) AviPoolBuild(pool_meta *nodes.AviPoolNode, cache_obj
 		pool.Servers = append(pool.Servers, &s)
 	}
 
-	var hm string
-	if pool_meta.Protocol == utils.UDP {
-		hm = fmt.Sprintf("/api/healthmonitor/?name=%s", utils.AVI_DEFAULT_UDP_HM)
+	// overwrite with healthmonitors provided by CRD
+	if len(pool_meta.HealthMonitors) > 0 {
```

Comment:
does it make sense to move this entire piece of code into layer 2 and ensure all transition cases are gracefully handled?

Reply:
This'll have to then go in several other model generation areas/translators.